pallets/rewards/src/lib.rs

```diff
@@ -0,0 +1,494 @@
+pub use pallet::*;
+
+/// A type alias for the balance type from this pallet's point of view.
+type BalanceOf<T> =
+	<<T as Config>::NativeCurrency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(test)]
+mod mock;
+
+mod crowdloan_rewardees;
+
+const MIN_REWARDS_CLAIMABLE_AMOUNT: u128 = polkadex_primitives::UNIT_BALANCE;
+pub const REWARDS_LOCK_ID: LockIdentifier = *b"REWARDID";
+// Definition of the pallet logic, to be aggregated at runtime definition through
+// `construct_runtime`.
+#[allow(clippy::too_many_arguments)]
+#[frame_support::pallet]
+pub mod pallet {
+	use core::fmt::Debug;
+	// Import various types used to declare pallet in scope.
+	use super::*;
+	use frame_support::{
+		pallet_prelude::{OptionQuery, *},
+		traits::{Currency, LockableCurrency, ReservableCurrency, WithdrawReasons},
+		PalletId,
+	};
+	use frame_system::pallet_prelude::*;
+	use sp_runtime::traits::{IdentifyAccount, Verify};
+	use sp_std::{cmp::min, convert::TryInto};
+	/// Our pallet's configuration trait. All our types and constants go in here. If the
+	/// pallet is dependent on specific other pallets, then their configuration traits
+	/// should be added to our implied traits list.
+	///
+	/// `frame_system::Config` should always be included.
+	#[pallet::config]
+	pub trait Config: frame_system::Config + timestamp::Config {
+		/// The overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// Address which holds the customer funds.
+		#[pallet::constant]
+		type PalletId: Get<PalletId>;
+
+		/// Balances Pallet
+		type NativeCurrency: Currency<Self::AccountId>
+			+ ReservableCurrency<Self::AccountId>
+			+ LockableCurrency<Self::AccountId>;
+
+		type Public: Clone
+			+ PartialEq
+			+ IdentifyAccount<AccountId = Self::AccountId>
+			+ Debug
+			+ parity_scale_codec::Codec
+			+ Ord
+			+ scale_info::TypeInfo;
+
+		/// A matching `Signature` type.
+		type Signature: Verify<Signer = Self::Public>
+			+ Clone
+			+ PartialEq
+			+ Debug
+			+ parity_scale_codec::Codec
+			+ scale_info::TypeInfo;
+
+		/// Governance Origin
+		type GovernanceOrigin: EnsureOrigin<<Self as frame_system::Config>::Origin>;
+	}
+
+	// Simple declaration of the `Pallet` type. It is placeholder we use to implement traits and
+	// method.
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// The extrinsic will be used to start a new reward cycle
+		/// # Parameters
+		/// * `origin`: The donor who wants to start the reward cycle
+		/// * `start_block`: The block from which reward distribution will start
+		/// * `end_block`: The block at which last rewards will be distributed
+		/// * `initial_percentage`: The percentage of rewards that can be claimed at start block
+		/// * `reward_id`: The reward id
+		#[pallet::weight(10_000)]
```

Comment:
No weights through benchmarks

Reply:
Issue  #629 opened for it.

Reply:
can we wait with merging this PR until benchmarks and weights are merged here?